week3/ex2_unique_generator.py

```diff
@@ -1,10 +1,8 @@
 
 def unique(mylist):
     dict_unique = {}
-    unique_values = [dict_unique.setdefault(e, e) for e in mylist if e not in dict_unique]
-
-    for v in unique_values:
-        yield v
+    for el in [dict_unique.setdefault(e, e) for e in mylist if e not in dict_unique]:
```

Comment:
The intermediate list is still there. 

```python
[dict_unique.setdefault(e, e) for e in mylist if e not in dict_unique]
```
This expression creates a list.
As a hint, try to rewrite this piece of code with usual for loop (not using list comprehensions).
The code should look like:
```python
for ....:
   ....
   yield ....
```

Reply:
done